File: Scripts/server/call-an-object-method/r/call-an-object-method.r
CallAnObjectMethod <- function(a) {
	value <- list(name = a)
	attr(value, "class") <- "CallAnObjectMethod"
}

speak <- function(obj) {
	UseMethod("speak")
}

speak.CallAnObjectMethod <- function(obj) {
	value = obj$name
}

executeTask <- function(i) {

	object <- list(name="Object")
	class(object) <- c("Object","CallAnObjectMethod")
	speak(object)
	return(i)
}
 r = 1
for ( i in 0:2000000000 ) {
	r = executeTask(i)
}

File: Scripts/server/call-an-object-method/javascript/call-an-object-method.js
class CallAnObjectMethod {
	constructor() {
	}

	someMethod(a) {
		var value = a;
	}
}

var r = 1;
const obj = new CallAnObjectMethod();

function executeTask(i) {

	obj.someMethod(1);
	return i + 1;
}

for (var i = 0; i < 2000000000; ++i) {
	r = executeTask(i + r);
}
